src/core/transaction_hash/starknet_transaction_hash.rs

```diff
@@ -116,6 +122,41 @@ pub fn calculate_deploy_account_transaction_hash(
     )
 }
 
+pub(crate) fn calculate_declare_transaction_hash(
+    contract_class: ContractClass,
+    chain_id: Felt,
+    sender_address: Address,
+    max_fee: u64,
+    version: u64,
+    nonce: Felt,
+) -> Result<Felt, SyscallHandlerError> {
+    let class_hash =
+        compute_class_hash(&contract_class).map_err(|_| SyscallHandlerError::FailToComputeHash)?;
+
+    let (calldata, additional_data) = if version > u64::pow(2, 63) {
```

Comment:
The compiler is probably smart enough to know that `u64::pow(2, 63)` is constant, but maybe `0x8000_0000_0000_0000` would be better?

Reply:
we are unsure about this condition, maybe it is not necessary. But i will change it as you suggest.